cmd/cstor-pool-mgmt/pool/v1alpha2/import.go

```diff
@@ -30,27 +31,55 @@ import (
 // It will return -
 // - If pool is imported or not
 // - If any error occurred during import operation
-func Import(csp *apis.CStorPoolInstance) (bool, error) {
-	if poolExist := checkIfPoolPresent(PoolName(csp)); poolExist {
+func Import(cspi *apis.CStorPoolInstance) (bool, error) {
+	if poolExist := checkIfPoolPresent(PoolName(cspi)); poolExist {
 		return true, nil
 	}
 
 	// Pool is not imported.. Let's update the syncResource
+	var cmdOut []byte
+	var err error
 	common.SyncResources.IsImported = false
+	var poolImported bool
 
-	if ret, er := zfs.NewPoolImport().
-		WithCachefile(csp.Spec.PoolConfig.CacheFile).
-		WithProperty("cachefile", csp.Spec.PoolConfig.CacheFile).
-		WithDirectory(SparseDir).
-		WithDirectory(DevDir).
-		WithPool(PoolName(csp)).
-		Execute(); er != nil {
-		glog.Errorf("Failed to import pool : %s : %s", ret, er.Error())
+	bdPath, err := getPathForBDev(cspi.Spec.RaidGroups[0].BlockDevices[0].BlockDeviceName)
+	if err != nil {
+		return false, err
+	}
+
+	glog.Infof("Importing pool %s %s", string(cspi.GetUID()), PoolName(cspi))
+	devID := pool.GetDevPathIfNotSlashDev(bdPath)
+	if len(devID) != 0 {
+		cmdOut, err = zfs.NewPoolImport().
+			WithCachefile(cspi.Spec.PoolConfig.CacheFile).
+			WithProperty("cachefile", cspi.Spec.PoolConfig.CacheFile).
+			WithDirectory(devID).
+			WithPool(PoolName(cspi)).
+			Execute()
+		if err == nil {
+			poolImported = true
+		} else {
+			// If pool import failed, fallback to try for import without Directory
+			glog.Errorf("Failed to import pool with directory %s : %s : %s",
+				devID, cmdOut, err.Error())
+		}
+	}
+
+	if poolImported == false {
+		cmdOut, err = zfs.NewPoolImport().
+			WithCachefile(cspi.Spec.PoolConfig.CacheFile).
+			WithProperty("cachefile", cspi.Spec.PoolConfig.CacheFile).
+			WithPool(PoolName(cspi)).
+			Execute()
+	}
+
+	if err != nil {
+		glog.Errorf("Failed to import pool : %s : %s", cmdOut, err.Error())
 		// We return error as nil because pool doesn't exist
```

Comment:
Please remove this comment since you are returning an error.

Reply:
IMO we should not return an error here since pool may not exist.

Reply:
removed the comment